server/services/team/service.ts

```diff
@@ -146,6 +149,112 @@ export class TeamService extends ServiceClass implements ITeamService {
 		};
 	}
 
+	async addRoom(uid: string, rid: string, teamId: string, isDefault: boolean): Promise<IRoom> {
+		if (!teamId) {
+			throw new Error('missing-teamId');
+		}
+		if (!rid) {
+			throw new Error('missing-roomId');
+		}
+		if (!uid) {
+			throw new Error('missing-userId');
+		}
+		const room = await this.RoomsModel.findOneByRoomIdAndUserId(rid, uid);
+		if (!room) {
+			throw new Error('invalid-room');
+		}
+		const team = await this.TeamModel.findOneById(teamId);
+		if (!team) {
+			throw new Error('invalid-team');
+		}
+		if (room.teamId) {
+			throw new Error('room-already-on-team');
+		}
+		if (isDefault == null) {
+			isDefault = false;
+		}
+		room.teamId = teamId;
+		room.teamDefault = isDefault;
+		this.RoomsModel.setTeamById(room._id, teamId, isDefault);
+		return {
+			...room,
+		};
+	}
+
+	async removeRoom(uid: string, rid: string, teamId: string): Promise<IRoom> {
+		if (!teamId) {
+			throw new Error('missing-teamId');
+		}
+		if (!rid) {
+			throw new Error('missing-roomId');
+		}
+		if (!uid) {
+			throw new Error('missing-userId');
+		}
+		const room = await this.RoomsModel.findOneByRoomIdAndUserId(rid, uid);
+		if (!room) {
+			throw new Error('invalid-room');
+		}
+		const team = await this.TeamModel.findOneById(teamId);
+		if (!team) {
+			throw new Error('invalid-team');
+		}
+		if (room.teamId !== teamId) {
+			throw new Error('room-not-on-that-team');
+		}
+		delete room.teamId;
+		delete room.teamDefault;
+		this.RoomsModel.unsetTeamById(room._id);
+		return {
+			...room,
+		};
+	}
+
+	async updateRoom(uid: string, rid: string, isDefault: boolean): Promise<IRoom> {
+		if (!rid) {
+			throw new Error('missing-roomId');
+		}
+		if (!uid) {
+			throw new Error('missing-userId');
+		}
+		const room = await this.RoomsModel.findOneByRoomIdAndUserId(rid, uid);
+		if (!room) {
+			throw new Error('invalid-room');
+		}
+		if (!room.teamId) {
+			throw new Error('room-not-on-team');
+		}
+		room.teamDefault = isDefault;
+		this.RoomsModel.setTeamDefaultById(rid, isDefault);
+		return {
+			...room,
+		};
+	}
+
+	async listRooms(uid: string, teamId: string, getAllRooms: boolean, allowPrivateTeam: boolean, { offset, count }: IPaginationOptions = { offset: 0, count: 50 }): Promise<IRecordsWithTotal<IRoom>> {
+		if (!teamId) {
+			throw new Error('missing-teamId');
+		}
+		const team = await this.TeamModel.findOneById(teamId, {});
+		if (!team) {
+			throw new Error('invalid-team');
+		}
+		const isMember = await this.TeamMembersModel.findOneByUserIdAndTeamId(uid, teamId);
+		if (team.type === TEAM_TYPE.PRIVATE && !allowPrivateTeam && !isMember) {
+			throw new Error('user-not-on-private-team');
+		}
+		let cursor;
+		if (getAllRooms) {
+			cursor = this.RoomsModel.findByTeamId(teamId, { limit: count, skip: offset });
+		} else {
+			cursor = this.RoomsModel.findByTeamIdAndUserId(uid, teamId, { limit: count, skip: offset });
```

Comment:
as I mentioned on the Rooms model, this action is actually querying room's creator, but instead I'd say you can do:

- get the list of rooms from the team
- check which rooms of those rooms the user has a *Subscription* (for example: `Subscription.find({ 'u._id': uid, rid: { $in: listOfRoomIdsFromTeam } })`)
- return only rooms returned from previous query